### PR TITLE
theme: Implement DesignVariables.dark

### PR DESCRIPTION
**Pull request by chrisbobbe**

### CODE REVIEW

File: lib/widgets/theme.dart [hunk 1]
```diff
@@ -75,19 +75,33 @@ const kZulipBrandColor = Color.fromRGBO(0x64, 0x92, 0xfe, 1);
...
 class DesignVariables extends ThemeExtension<DesignVariables> {
-  DesignVariables() :
-    bgMain = const Color(0xfff0f0f0),
-    bgTopBar = const Color(0xfff5f5f5),
-    borderBar = const Color(0x33000000),
-    icon = const Color(0xff666699),
-    title = const Color(0xff1a1a1a),
-    streamColorSwatches = StreamColorSwatches.light;
+  factory DesignVariables.light() {
+    return DesignVariables._(
```
Comment (gnprice) [inline lib/widgets/theme.dart:79]: nit: can simplify slightly:
```suggestion
  DesignVariables.light() :
    this._(
```

That's a Dart "redirecting constructor":
https://dart.dev/language/constructors#redirecting-constructors

**Review by gnprice — COMMENTED** [review]
Exciting!

Let's take this as a prompt to add a quick set of smoke-tests of the `lerp` method here, from light to dark etc. to check that they don't blow up, same as was done recently for CodeBlockTextStyles.

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed. I've also added some stream-colors test coverage that's meant to be added here:

```
// TODO(#95) test with Brightness.dark and lerping between light/dark
```

File: test/widgets/theme_test.dart [hunk 2]
```diff
@@ -76,30 +76,71 @@ void main() {
...
+      final subscription = eg.subscription(eg.stream(), color: baseColor);
 
-        late StreamColorSwatch actualSwatch;
-        final navigator = await ZulipApp.navigator;
-        navigator.push(MaterialWidgetRoute(page: Builder(builder: (context) {
-          actualSwatch = colorSwatchFor(context, subscription);
-          return const Placeholder();
-        })));
-        await tester.pumpAndSettle();
+      debugFollowPlatformBrightness = true;
```
Comment (gnprice) [inline test/widgets/theme_test.dart:115]: should also get an `addTearDown` call to reset

File: test/widgets/theme_test.dart [hunk 3]
```diff
@@ -76,30 +76,71 @@ void main() {
...
+      final expectedLerped = StreamColorSwatch.lerp(
+        StreamColorSwatch.light(baseColor),
+        StreamColorSwatch.dark(baseColor),
+        0.4)!;
+      check(colorSwatchFor(element, subscription)).equals(expectedLerped);
 
-    doTest('light', 0xff76ce90, StreamColorSwatch.light(0xff76ce90));
-    // TODO(#95) test with Brightness.dark and lerping between light/dark
+      await tester.pump(kThemeAnimationDuration * 0.6);
+      check(colorSwatchFor(element, subscription)).equals(StreamColorSwatch.dark(baseColor));
```
Comment (gnprice) [inline test/widgets/theme_test.dart:143]: nit: wrap line, also making parallel to the light case above
```suggestion
      check(colorSwatchFor(element, subscription))
        .equals(StreamColorSwatch.dark(baseColor));
```

Then might as well make the intermediate case parallel too: inline `expectedLerped`, and format the same way, like
```dart
      check(colorSwatchFor(element, subscription))
        .equals(StreamColorSwatch.lerp(
```

File: test/widgets/theme_test.dart [hunk 4]
```diff
@@ -76,30 +76,71 @@ void main() {
...
+      await tester.pump();
+
+      final navigator = await ZulipApp.navigator;
+      navigator.push(MaterialWidgetRoute(page: Builder(builder: (context) =>
+        const Placeholder())));
+      await tester.pumpAndSettle();
+
+      final element = tester.element(find.byType(Placeholder));
+      // Compares all the swatch's members; see [ColorSwatch]'s `operator ==`.
+      check(colorSwatchFor(element, subscription))
```
Comment (gnprice) [inline test/widgets/theme_test.dart:129]: Neat!

**Review by gnprice — COMMENTED** [review]
Thanks! Two small comments; then please go ahead and merge.

**Comment by chrisbobbe** [comment]
Thanks! Done.

**Pull request merged**